cli/cmd/upgrade.go

```diff
@@ -156,6 +156,15 @@ func makeUpgradeFlags() *pflag.FlagSet {
 
 func upgradeRunE(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, crds bool, options valuespkg.Options) error {
 
+	if !crds {
+		err := healthcheck.CheckCustomResourceDefinitions(ctx, k)
+		if err != nil {
+			fmt.Println(err)
+			fmt.Fprintln(os.Stderr, "Linkerd CRDs must be installed first. Run linkerd upgrade with the --crds flag.")
+			os.Exit(1)
```

Comment:
I don't think `println` makes sense here -- we'll send the error message to output stream, when `upgrade` generates yaml. This means that a command like `linkerd upgrade > linkerd.yaml` could populate the file with non-yaml data if we hit this, for instance.

And I'm pretty sure the caller prints the error message above, so we could do something like
```suggestion
			return fmt.Errorf("Linkerd CRDs must be installed first. Run linkerd upgrade with the --crds flag:\n%w", err)
```